octobot_trading/data/order.py

```diff
@@ -258,16 +217,38 @@ def get_total_fees(self, currency):
         return get_fees_for_currency(self.fee, currency)
 
     def is_open(self):
-        return self.status in {OrderStatus.OPEN, OrderStatus.PARTIALLY_FILLED}
+        return self.state.is_open()
 
     def is_filled(self):
-        return self.status is OrderStatus.FILLED
+        return self.state.is_filled() or (self.state.is_closed() and self.status is OrderStatus.FILLED)
 
     def is_cancelled(self):
-        return self.status is OrderStatus.CANCELED
+        return self.state.is_canceled() or (self.state.is_closed() and self.status is OrderStatus.CANCELED)
 
     def is_closed(self):
-        return self.status in {OrderStatus.CANCELED, OrderStatus.FILLED, OrderStatus.CLOSED}
+        return self.state.is_closed() if self.state is not None else self.status == OrderStatus.CLOSED
```

Comment:
a filled order is not considered closed ?

Reply:
A filled Order is automatically considered as close just after filling complete